src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -171,6 +171,42 @@ def get_component_contexts(
                 output.append(context_item)
         return output
 
+    def get_installation_component_contexts(
+        self,
+        *,
+        filter: SentryAppInstallationFilterArgs,
+        component_type: str,
+        include_contexts_without_component: bool = False,
+    ) -> list[RpcSentryAppComponentContext]:
+        install_query = self._FQ.query_many(filter=filter)
+        app_ids = install_query.select_related("sentry_app", "sentry_app__application").values_list(
+            "sentry_app_id", flat=True
+        )
```

Comment:
If you're only reading `sentry_app_id` from this result set, do you need the joins to `sentry_app` and `sentry_app__application`?